modules/instruments/php/module.class.inc

```diff
@@ -146,5 +146,37 @@ class Module extends \Module
         }
         return [];
     }
+
+    /**
+     * Return the data dictionary for all instruments installed on a LORIS
+     * instance.
+     *
+     * @param \LORIS\LorisInstance $loris The loris instance whose dictionary
+     *                                    should be retrieved
+     *
+     * @return \LORIS\Data\Dictionary\Category[]
+     */
+    public function getDataDictionary(\LORIS\LorisInstance $loris) : iterable
+    {
+        $DB = $loris->getDatabaseConnection();
+
+        $rows = $DB->pselectCol("SELECT Test_name FROM test_names", []);
```

Comment:
Should probably use getAllInstruments instead (and fix it to query the proper Full name)
https://github.com/aces/Loris/blob/main/php/libraries/Utility.class.inc#L459

Reply:
It is using the full name: https://github.com/aces/Loris/pull/7179/files#diff-e52203453b6573a6a78d0ecbf197ba50a8a893a8d70bb452dac8788d8d19b9ddR171 .. the full name from that Utility function is unreliable. It would be incorrect for ie. LINST instruments, or any instrument not using LegacyInstrumentTrait. Do you still want me to change it to call the utlity function/use array_keys on it to get the test_name for the factory call? I can, but it doesn't seem worthwhile.

Reply:
No i was thinking modifying the utility function to get the correct Full names instead

Reply:
I changed it to use the Utility function to avoid the database call, but I don't want to change the behaviour of the Utility function in this PR since it's (probably?) used other places in LORIS

Reply:
https://github.com/aces/Loris/pull/7398